src/integrations/checkpoints/CheckpointTracker.ts

```diff
@@ -157,7 +157,10 @@ class CheckpointTracker {
 
 			console.info(`Using shadow git at: ${gitPath}`)
 
-			await this.gitOperations.addCheckpointFiles(git)
+			const addFilesResult = await this.gitOperations.addCheckpointFiles(git)
+			if (!addFilesResult.success) {
+				console.error("Failed to add at least one file(s) to checkpoints shadow git")
```

Comment:
should we not be breaking out of execution here by adding `return`? If the add files fails it proceeds as normal?